### PR TITLE
Lower `index_copy`

### PR DESCRIPTION
**Pull request by miladm**
Lower `index_copy` in PyTorch/XLA.

This issue is to enable Structured Kernels for `index_copy` in PyTorch Lazy Tensor Core (LRC). Here is the reference PyTorch [PR](https://github.com/pytorch/pytorch/pull/67329) this change unblocks.

### CODE REVIEW

**Comment by miladm** [comment]
@badenh, in this [PR] you point out XLA must only lower `index_copy`, meaning we can drop `index_copy_`. I added both in my first PR push. Please confirm `index_copy` alone is sufficient.

Thanks!

**Comment by bdhirsh** [comment]
I'd expect that you can drop `index_copy_` and only lower `index_copy` - the codegen should generate an `index_copy_` and `index_copy(out=...)` lowering for you.

You also might want to add a TORCH_PIN to the PR from core, so we can make sure CI is green (https://github.com/pytorch/pytorch/pull/67329)

**Review by badenh — APPROVED** [review]

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -1615,14 +1615,13 @@ at::Tensor XLANativeFunctions::index_add(const at::Tensor& self, int64_t dim,
       bridge::GetXlaTensor(source), alpha));
 }
 
-at::Tensor& XLANativeFunctions::index_copy_(at::Tensor& self, int64_t dim,
+at::Tensor& XLANativeFunctions::index_copy(const at::Tensor& self, int64_t dim,
                                             const at::Tensor& index,
                                             const at::Tensor& source) {
   XLA_FN_COUNTER("xla::");
   XLATensor self_tensor = bridge::GetXlaTensor(self);
-  XLATensor::index_copy_(self_tensor, dim, bridge::GetXlaTensor(index),
```
Comment (bdhirsh) [inline torch_xla/csrc/aten_xla_type.cpp:1623]: nit: you can probably kill the code for `XLATensor::index_copy_()` too?

**Comment by miladm** [comment]
Blocked on CI test error below. I believe is about to go in any time today. Thanks @wonjoolee95.

```
1 out of 3 hunks FAILED
Applying patch file: /tmp/pytorch/xla/tf_patches/multi_load_libtpu.diff
Traceback (most recent call last):
  File "/tmp/pytorch/xla/scripts/cond_patch.py", line 67, in <module>
    patch_repo(args)
  File "/tmp/pytorch/xla/scripts/cond_patch.py", line 49, in patch_repo
    raise RuntimeError('Failed to apply patch: {}'.format(ppath))
RuntimeError: Failed to apply patch: /tmp/pytorch/xla/tf_patches/multi_load_libtpu.diff
```

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (miladm) [reply torch_xla/csrc/aten_xla_type.cpp:1623]: @bdhirsh I believe that's what I did. This change no longer has the call to `torch_copy_`. Is the way it's done incorrect for you?

Comment (bdhirsh) [reply torch_xla/csrc/aten_xla_type.cpp:1623]: I just meant that there's also a `index_copy_()` static method on the `XLATensor` class (in `torch_xla/csrc/tensor.h` that should be unused now. So I think we can remove that method too?

Comment (miladm) [reply torch_xla/csrc/aten_xla_type.cpp:1621]: That's correct. I understand what you meant. 
FYI: I am fixing a local build issue in my dev env. Will push the final changes soon.

**Review by badenh — APPROVED** [review]
Thanks for handling all the feedback

**Review by bdhirsh — APPROVED** [review]
LGTM, thanks Milad :).

Got a minor fix to make on https://github.com/pytorch/pytorch/pull/67329/files, then I'll land them

**Comment by bdhirsh** [comment]
This PR is ready to land - the core side change has landed (https://github.com/pytorch/xla/pull/3331)

**Pull request merged**